great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -2794,16 +2794,32 @@ def add_or_update_expectation_suite(  # noqa: PLR0913
                 execution_engine_type=execution_engine_type,
                 meta=meta,
             )
+        existing = None
+        if id is not None:
+            try:
+                existing = self.get_expectation_suite(ge_cloud_id=id)
+            except gx_exceptions.DataContextError:
```

Comment:
Is there an error code we can use (or anything more specific), to ensure this error is actually about the suite not being found? It doesn't seem like we should need to call `get_expectation_suite`. Mercury should return an error code that is useful.